source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -224,25 +230,35 @@ bool SimpleHttpCache::insert(const Key& key, Http::ResponseHeaderMapPtr&& respon
   return true;
 }
 
-SimpleHttpCache::Entry
-SimpleHttpCache::varyLookup(const LookupRequest& request,
-                            const Http::ResponseHeaderMapPtr& response_headers) {
-  // This method should be called from lookup, which holds the mutex for reading.
-  mutex_.AssertReaderHeld();
-
+absl::optional<Key>
+SimpleHttpCache::variedRequestKey(const LookupRequest& request,
```

Comment:
variedRequestKey doesn't use any members of SimpleHttpCache, so there's no reason for it to be a member function. It would be better as either a file-scoped static or unnamed namespace member, or in VaryHeaderUtils.

Reply:
Tried to move it to VaryHeaderUtils but that creates a circular dependency for LookupRequest, or requires splitting that parameter into an ugly three parameters which would make it sensible to add a helper wrapper in this file, at which point it's not really worth the extra indirection, so file-scoped static it is.